### PR TITLE
Create enterText-trailing-caret.md

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
pending PR: https://github.com/flutter/flutter/pull/79506

### CODE REVIEW

File: src/docs/release/breaking-changes/enterText-trailing-caret.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+`WidgetTester.enterText` and `TestTextInput.enterText` now set the 
+selection to `TextSelection.collapsed(offset: text.length)`, instead
+of `TextSelection.collapsed(offset: -1)`.
+
+## Migration guide
+
+It should be very uncommon for tests to have to rely on the 
+previous behavior of `enterText`, since usually the selection 
+should not be invalid. **Consider changing the expected values of 
+your tests to adopt the `enterText` change.**
```
Comment (Hixie) [inline src/docs/release/breaking-changes/enterText-trailing-caret.md:39]: you should include examples of error messages you might get here, so that googling for them reaches this page.

i would have expected a common failure to be golden file tests.

File: src/docs/release/breaking-changes/enterText-trailing-caret.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+If your tests have to relay on setting the selection to invalid,
```
Comment (Hixie) [inline src/docs/release/breaking-changes/enterText-trailing-caret.md:41]: rely, not relay

File: src/docs/release/breaking-changes/enterText-trailing-caret.md (hunk 1 above)
Comment (LongCatIsLooong) [reply src/docs/release/breaking-changes/enterText-trailing-caret.md:39]: Any keywords I should add to help people reach this page via googling? I feel it won't be particularly easy to find this by googling "golden test failure" or "caret moved to the end".

File: src/docs/release/breaking-changes/enterText-trailing-caret.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+---
+title: `enterText` to move the caret to the end of the input text.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/enterText-trailing-caret.md:2]: Can you rephrase? Something like:

```
title: Change the enterText method to move the caret to the end of the input text
```

We don't use codefont in a title.

File: src/docs/release/breaking-changes/enterText-trailing-caret.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+## Summary
+
+`WidgetTester.enterText` and `TestTextInput.enterText` now move the
+caret to the end of the input text.
+
+## Context
+
+The caret indicates the insertion point within the current text in an 
+active input field. Typically when a new character is entered, the
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/enterText-trailing-caret.md:14]:
```suggestion
active input field. Typically, when a new character is entered, the 
```

**Review by sfshaza2 — COMMENTED** [review]
This LGTM with a couple minor nits.

**Review by sfshaza2 — APPROVED** [review]
LGTM
